railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -295,13 +300,23 @@ func (p *clientgoCachedProxyResolver) updateKongAdmin() error {
 	// generate the deck configuration to be applied to the admin API
 	targetConfig := deckgen.ToDeckContent(p.ctx, p.deprecatedLogger, kongstate, nil, nil)
 
+	// retrieve custom entities
+	var customEntities []byte
+	if p.kongConfig.InMemory && p.KongCustomEntitiesSecret != "" {
+		customEntities, err = fetchCustomEntities(p.KongCustomEntitiesSecret, storer)
+		if err != nil {
+			// failure to fetch custom entities shouldn't block updates
+			p.logger.Error(err, "failed to fetch custom entities")
```

Comment:
If the error is a cache miss, we can just info or even debug log that right? "No custom entities secret found, no big deal", and then hard fail on anything else because other errors are unexpected?

Reply:
Ditto https://github.com/Kong/kubernetes-ingress-controller/pull/1339#discussion_r638311938. This is how we fail currently; no pressing reason to break from the 1.x behavior.